addons/website/models/ir_module_module.py

```diff
@@ -549,3 +550,184 @@ def _load_module_terms(self, modules, langs, overwrite=False):
             """, (default_menu.id,))
 
         return res
+
+    def _generate_primary_snippet_templates(self):
+        def split_key(snippet_key):
+            # Snippets xmlid can be written without the module part, meaning it is
+            # a shortcut for a website module snippet.
+            return snippet_key.split('.') if '.' in snippet_key else ('website', snippet_key)
+
+        def create_model_data(missing_records):
+            # Creates the model data records for newly created template records.
+            self.env['ir.model.data'].create([{
+                'name': record.key.split('.')[1],
+                'module': record.key.split('.')[0],
+                'model': 'ir.ui.view',
+                'res_id': record.id,
+                'noupdate': False,
+            } for record in missing_records])
+
+        def create_missing(create_values):
+            # Creates the snippet primary template records that do not exist yet.
+            # Safety net in case something goes wrong, it should not be possible
+            # but it is a low effort to do it.
+            create_values = [values for values in create_values if values]
+            keys = [values['key'] for values in create_values]
+            existing_primary_template_keys = self.env['ir.ui.view'].search_fetch([('mode', '=', 'primary'), ('key', 'in', keys)], ['key']).mapped('key')
+            missing_create_values = [values for values in create_values if values['key'] not in existing_primary_template_keys]
+            missing_records = self.env['ir.ui.view'].with_context(no_cow=True).create(missing_create_values)
+            create_model_data(missing_records)
+            # Prevent deletion by _process_end
+            for values in create_values:
+                self.env['ir.model.data']._load_xmlid(values["key"])
+            return len(missing_records.ids)
+
+        def get_create_vals(name, snippet_key, parent_wrap, new_wrap):
+            # Returns the create values for the new primary template of the
+            # snippet having snippet_key as its base key, having a new key
+            # formatted with new_wrap, and extending a parent with the key
+            # formatted with parent_wrap.
+            mod, xmlid = split_key(snippet_key)
+            parent_key = f'{mod}.{parent_wrap % xmlid}'
+            parent_id = self.env['ir.model.data']._xmlid_to_res_model_res_id(parent_key, False)
+            if not parent_id:
+                _logger.warning('No such snippet template: %s', parent_key)
+                return None
+            return {
+                'name': name,
```

Comment:
Adding this here:
![image](https://github.com/odoo/odoo/assets/30048408/df1d0e41-ac2d-4289-b1e3-226904680e20)

Some "siblings" templates have different name formatting, some with name being the same as the key, some being sentence with words.
Eg
![image](https://github.com/odoo/odoo/assets/30048408/f7607f15-dedc-4c6b-b2a7-f72232ea6c8c)

One is the pricing category, the other the about category, same level, why different name formatting?

Also, about this one
![image](https://github.com/odoo/odoo/assets/30048408/2142c73b-6961-41a7-a8d0-4477a7e42fc8)
It seems like it only created MAP and FULL for ABOUT, while it has full_1, mini, personnal and timeline too, is this normal?

Reply:
About which ones are created: the manifest only mentions `s_text_image` in the MAP and FULL for ABOUT - therefore only those templates are created (it was to avoid prebuilding 80K+ templates by default)

About the naming: the full text ones are the generated ones, the others are the ones that are created explictly in the XML because they patch the content. We did not specify the names there.